### PR TITLE
PlayButton Updates & Minor Card Cleanup

### PR DESCRIPTION
**Pull request by tayiorbeii**
image

# Changes:
* Add different `hover` classes to `<PlayButton />` depending on Card type
* Clean up styling based on Card type
* DRY out `expanded` prop type checking
* Strengthen `playlist` Card styling check

image

### CODE REVIEW

**Review by joelhooks — APPROVED** [review]

File: src/components/Button/PlayButton.js [hunk 1]
```diff
@@ -15,15 +26,32 @@ const StyledPlayButton = styled.div`
     position: relative;
     left: 3px;
   }
-  ${props => props.hover ? 'opacity: 0;' : ''},
+  ${props => props.hover ? 'opacity: 0;' : ''}
+  ${props => 
+    props.type === 'lesson' ||
+    (props.type === 'course' && (props.expanded === false || props.expanded === 'horizontal'))
+     ? `${centerButtonRules}`
```
Comment (tayiorbeii) [inline src/components/Button/PlayButton.js:33]: if it's a `LessonCard` or a "regular" `CourseCard` or a Horizontal `CourseCard`, apply the CSS rules in `centerButtonRules` (line 14)

File: src/components/Button/PlayButton.js [hunk 2]
```diff
@@ -15,15 +26,32 @@ const StyledPlayButton = styled.div`
...
+     : ''
+   }
 `
 
-const PlayButton = ({ hover=false, className }) => {
-  return <StyledPlayButton hover={hover} className={`${hover ? `${hoverPlayBtnClasses} card-play-btn` : playBtnClasses} ${className}`}/>
+const PlayButton = ({ hover=false, type, expanded, className }) => {
+  return <StyledPlayButton hover={hover}
+            className={`${hover
+                          ? `${hoverPlayBtnClasses} ${hoverColorClasses[type] ? hoverColorClasses[type] : ''} card-play-btn`
```
Comment (tayiorbeii) [inline src/components/Button/PlayButton.js:41]: if `hover` is true,  apply the `hoverPlayBtnClasses` for sure, and see if we have additional classes to apply by looking up the card `type` in `hoverColorClasses`....

File: src/components/Button/PlayButton.js [hunk 3]
```diff
@@ -15,15 +26,32 @@ const StyledPlayButton = styled.div`
...
+                          : playBtnClasses
```
Comment (tayiorbeii) [inline src/components/Button/PlayButton.js:42]: ...otherwise just apply the `playBtnClasses`

File: src/components/Button/PlayButton.js [hunk 4]
```diff
@@ -15,15 +26,32 @@ const StyledPlayButton = styled.div`
...
+                         }
+                        ${type === 'playlist' ? `${hoverColorClasses[type]}` : ''}
```
Comment (tayiorbeii) [inline src/components/Button/PlayButton.js:44]: `<PlaylistCard>`'s `PlayButton` is always visible, but it does get a color on hover

File: src/components/Button/PlayButton.js [hunk 5]
```diff
@@ -15,15 +26,32 @@ const StyledPlayButton = styled.div`
...
+                        ${type === 'course' && expanded === 'vertical' ? 'bottom-1' : ''}
```
Comment (tayiorbeii) [inline src/components/Button/PlayButton.js:45]: Vertical `CourseCard` gets the tachyon `bottom-1`

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "egghead-ui",
-  "version": "1.1.0",
+  "version": "1.4.0",
```
Comment (trevordmiller) [inline package.json:3]: I would recommend not versioning until after a PR is approved and merged into master; so a release is separate from the feature PR. Here is my recommended workflow - would be good to add to the project README if you agree:

### Make changes
1) checkout and pull latest from master
2) create feature branch and check it out
3) make changes
4) create PR
5) once PR is approved, merge into master

### Release new version
6) checkout `release` and pull latest from master
7) run `yarn release`
8) merge release into master

### Notify consumers
9) consumers can then run `yarn upgrade egghead-ui` in their projects

Although the checkout in step 6 and step 8 wouldn't be needed if you could release from master

**Review by trevordmiller — APPROVED** [review]
Nice work Evgeniy and Taylor

**Review by ltclmbdev — APPROVED** [review]

**Pull request merged**